src/apps/vpn/serverlatency.cpp

```diff
@@ -225,3 +270,36 @@ void ServerLatency::recvPing(quint16 sequence) {
 void ServerLatency::criticalPingError() {
   logger.info() << "Encountered Unrecoverable ping error";
 }
+
+unsigned int ServerLatency::avgLatency() const {
+  if (m_latency.isEmpty()) {
+    return 0;
+  }
+  return (m_sumLatencyMsec + m_latency.count() - 1) / m_latency.count();
+}
+
+double ServerLatency::progress() const {
+  if ((m_pingSender == nullptr) || (m_pingSendTotal == 0)) {
+    return 1.0;  // Operation is complete.
+  }
+
+  double remaining = m_pingReplyList.count() + m_pingSendQueue.count();
+  return 1.0 - (remaining / m_pingSendTotal);
+}
+
+void ServerLatency::setCooldown(const QString& publicKey, qint64 timeout) {
+  if (timeout <= 0) {
+    m_cooldown.remove(publicKey);
+  } else {
+    m_cooldown[publicKey] = QDateTime::currentSecsSinceEpoch() + timeout;
+  }
+
+  // Emit signals that the connection score may have changed.
+  ServerCountryModel* scm = MozillaVPN::instance()->serverCountryModel();
+  const Server& server = scm->server(publicKey);
+  const ServerCity& city =
+      scm->findCity(server.countryCode(), server.cityName());
+  if (city.initialized()) {
+    emit city.scoreChanged();
```

Comment:
Sometimes instead of getting the "Last updated X ago" string, I get the default %1 placeholder. I am able to reproduce this if I hit refresh then turn the VPN on before the check completes. If user turns on the VPN mid-refresh we should probably just default back to the last "Last updated..." timestamp.

![Screen Recording 2023-02-09 at 10 15 21 AM](https://user-images.githubusercontent.com/3746552/217902323-e6630bbb-d85a-4aef-b918-0bf73a0bb5d6.gif)